lms/djangoapps/instructor_analytics/basic.py

```diff
@@ -84,7 +84,66 @@ def issued_certificates(course_key, features):
     return generated_certificates
 
 
-def enrolled_students_features(course_key, features):
+def get_student_features_with_custom(course_key):
+    """
+    Allow site operators to include on the export custom fields if platform has an extending
+    User model. This can be used if you have an extended model that include for example
+    an university student number.
+
+    Basic example of adding age:
+    ```python
+    def get_age(self):
+        return datetime.datetime.now().year - self.profile.year_of_birth
+    setattr(User, 'age', property(get_age))
+    ```
+    Then you have to add `age` to both site configurations:
+    - `student_profile_download_fields_custom_student_attributes`
+    - `student_profile_download_fields` site configurations`
+
+    ```json
+    "student_profile_download_fields_custom_student_attributes": ["age"],
+    "student_profile_download_fields": [
+        "id", "username", "name", "email", "language", "location",
+        "year_of_birth", "gender", "level_of_education", "mailing_address",
+        "goals", "enrollment_mode", "last_login", "date_joined", "external_user_key",
+        "enrollment_date", "age"
+    ]
+    ```
+
+    Example if the platform has a custom user extended model like a One-To-One Link
+    with the User Model:
+    ```python
+    def get_user_extended_model_custom_field(self):
+        if hasattr(self, "userextendedmodel"):
+            return self.userextendedmodel.custom_field
+        return None
+    setattr(User, 'user_extended_model_custom_field', property(get_user_extended_model_custom_field))
+    ```
+
+    ```json
+    "student_profile_download_fields_custom_student_attributes": ["user_extended_model_custom_field"],
+    "student_profile_download_fields": [
+        "id", "username", "name", "email", "language", "location",
+        "year_of_birth", "gender", "level_of_education", "mailing_address",
+        "goals", "enrollment_mode", "last_login", "date_joined", "external_user_key",
+        "enrollment_date", "user_extended_model_custom_field"
+    ]
+    ```
+    """
+    return STUDENT_FEATURES + tuple(
+        configuration_helpers.get_value_for_org(
+            course_key.org,
+            "student_profile_download_fields_custom_student_attributes",
+            getattr(
+                settings,
+                "STUDENT_PROFILE_DOWNLOAD_FIELDS_CUSTOM_STUDENT_ATTRIBUTES",
```

Comment:
Hi @igobranco , could you refactor this new feature flag to use documented edx-toggles instead? See [How to: Documenting new feature toggles](https://edx.readthedocs.io/projects/edx-toggles/en/latest/how_to/documenting_new_feature_toggles.html) for instructions, and [instructor.toggles](https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/instructor/toggles.py) and [discussion.config.settings](https://github.com/openedx/edx-platform/blob/3a343f8301f71878628ad4e36e6160bdbec4862d/lms/djangoapps/discussion/config/settings.py#L7) for examples.

Unfortunately I don't think there's a way to use SiteConfiguration as a toggle, so it'll need to be a DjangoSetting (feature flag), Waffle Switch or Waffle Flag. But if org-wide settings are a requirement for your use case, let's talk about it?

Reply:
Say someone uses `STUDENT_PROFILE_DOWNLOAD_FIELDS_CUSTOM_STUDENT_ATTRIBUTES = ['age']` but fails to add the property/method to the User model. When they run the profile report, [this line](https://github.com/openedx/edx-platform/pull/32216/files#diff-e74a04be02f2ed67823d2690541749f076e667da6160430d741d05be75ac2243R189) will throw a TypeError.

Could you please update `extract_attr(student, feature)` to catch that TypeError?

You could communicate the error in the report field if you think that's an appropriate place for it, but it should at least be logged, and shouldn't cause the whole report to fail.

Reply:
Hi @igobranco , do you know roughly when you'll have time to complete this PR? It would be great if we could get this feature into Redwood.

Reply:
Nudge @igobranco  😃

Reply:
Thanks for your updates @igobranco . Do you have any thoughts on my suggestions above?